packages/opentelemetry/src/spanExporter.ts

```diff
@@ -35,60 +35,104 @@ type SpanNodeCompleted = SpanNode & { span: ReadableSpan };
 const MAX_SPAN_COUNT = 1000;
 const DEFAULT_TIMEOUT = 300; // 5 min
 
+interface FinishedSpanBucket {
+  timestampInS: number;
+  spans: Set<ReadableSpan>;
+}
+
 /**
  * A Sentry-specific exporter that converts OpenTelemetry Spans to Sentry Spans & Transactions.
  */
 export class SentrySpanExporter {
   private _flushTimeout: ReturnType<typeof setTimeout> | undefined;
-  private _finishedSpans: ReadableSpan[];
-  private _timeout: number;
-
-  public constructor(options?: { timeout?: number }) {
-    this._finishedSpans = [];
-    this._timeout = options?.timeout || DEFAULT_TIMEOUT;
+  private _finishedSpanBuckets: (FinishedSpanBucket | undefined)[];
```

Comment:
l: Could we add comments to these explaining what they do here, e.g. why do we keep the spansToBucketEntry etc? it all makes sense to me, but for the sake of our future selves :D

Reply:
👍 added a comment